src/dali_executor/utils/utils.h

```diff
@@ -102,6 +102,27 @@ std::vector<TensorListShape<Dims>> split_list_shape(const TensorListShape<Dims>
   return result;
 }
 
+
+inline TensorListShape<> split_outer_dim(const TensorListShape<> &shape) {
+  auto in_ndim = shape.sample_dim();
```

Comment:
How about adding a test for this function?

Reply:
Done